plugins/gatsby-remark-args-linker/index.js

```diff
@@ -0,0 +1,74 @@
+const visit = require('unist-util-visit')
+
+const argsRegex = new RegExp(/\-{1,2}[a-zA-Z-]*/, 'ig')
+
+function patch(context, key, value) {
+  if (!context[key]) {
+    context[key] = value
+  }
+
+  return context[key]
+}
+const svgIcon = `<svg aria-hidden="true" focusable="false" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg>`
+
+module.exports = (
+  { markdownAST },
+  { icon = svgIcon, className = 'anchor', isIconAfterHeader = false }
+) => {
+  visit(
+    markdownAST,
+    node =>
+      node.type === 'listItem' &&
+      node.children.some(
```

Comment:
will it match lists like "- some text `--flag`"? also does it mean that we math lists outside the `Options` section?

Reply:
it matches list with 1st line start with inline Code block i.e `` and that starts with `-`.  Yes, If it exactly matches this then it's possible outside options too.  We can work on this more if it collide but I found that unique for my quick overlook over the docs.

Reply:
We do have a list that matches the pattern in our DVCLive docs: https://dvc-org-interactive-syn-zo8hwd.herokuapp.com/doc/dvclive/dvclive-with-dvc#--live-no-cache. Not sure if it does any harm to have links there... Though if we wanted, we could have `gatsby-remark-args-linker` only look through pages in our `/command-reference/` directory 🤔

Reply:
thanks, Julie and Thomas. q: how hard it is to make it strict and well-defined?

Reply:
The possible way I could think of was to make it strict or well defined is by checking heading Options and looking into a consecutive list and to the list item. 
But, I found it difficult to do so with Markdown AST. We could easily achieve that if we had something easy/well-defined APIs/functions as in DOM manipulation. Otherwise, we need to come up with some hacky way to do so.

Reply:
I looked into this a little earlier, simple checks like the dir being in `command-reference` can be done with parts of the Gatsby remark plugin API that we don't currently use: the plugins are actually passed the Markdown node of the page in question as well as the capability to reach into other nodes like its File parent.

```ts
module.exports = api => {
  const { markdownAST, getNode, markdownNode } = api
  const parentNode = getNode(markdownNode.parent)
  const isDoc = parentNode.relativeDirectory.startsWith(
    'docs/command-reference'
  )
```

More granular checks like ensuring the header text is "Options" would require dipping into the AST, which is certainly doable but the implementation can border on hacky- might be worth looking into in the future, but limiting this to `command-reference` pages should be enough limitation for a first iteration.

Doing either of these would require us to make this linker plugin live outside of the main `flow` statement that only passes in the markdown AST, but I think that's a convention worth breaking.

Reply:
sounds good, but it's also about replacing `node.children.some` with something like `node.children.first`

'docs/command-reference' - we need to make it a param? since it might be different across different websites later.

Reply:
> sounds good, but it's also about replacing node.children.some with something like node.children.first

This would check the first child in the list, right? While this would almost all use cases [`/repro`](https://dvc-org-interactive-syn-s47nhu.herokuapp.com/doc/command-reference/repro) options starts with `targets`:

![image](https://user-images.githubusercontent.com/43496356/148589463-1185de98-6cbe-4686-84a2-8ed5fd11bda7.png)

Is the only one as far as I know, though :)

Reply:
Oops sorry, I missed this conversation.

Reply:
I have added the params for paths which can be a string(path) or an array of paths. For now, I have set it to `docs/command-reference`.

Reply:
> sounds good, but it's also about replacing `node.children.some` with something like `node.children.first`

I have made some changes and pushed.